### PR TITLE
Use Percy to take snapshots in E2E tests

### LINKED ISSUE #2443: Percy Snapshots for E2E Tests
## Feature Description

We've began adding some E2E tests for critical flows that span beyond just the editor and involve a running WordPress instance.

With the increasing number of these tests, it would be nice if we could also add Percy snapshots to verify the flows also from a visual perspective.

## Additional Context

* [Percy + Puppeteer documentation](https://docs.percy.io/docs/puppeteer)
* Percy CI setup is currently disabled for Karma tests due to throttling issues. Let's also make sure it will initially be disabled for E2E tests until those issues are addressed.

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by swissspidy**
## Summary

Use Percy to take snapshots in E2E tests

## Relevant Technical Choices

Uses `@percy/puppeteer` to take snapshots in E2E tests, as per the [docs](https://docs.percy.io/docs/puppeteer).

## To-do

* [x] Use different Percy project
* [ ] Fix viewport discrepancies
* [ ] Make sure elements are not inserted at random positions

## User-facing changes

N/A

## Testing Instructions

Check screenshots on Percy.io.

---

Fixes #2443

### CODE REVIEW

**Comment by spacedmonkey** [comment]
How this connect and upload to percy? Can I see an example of this working?

**Comment by swissspidy** [comment]
The `percySnapshot` function does all the magic of taking & uploading screenshots.

These can then be reviewed at https://percy.io/Web-Stories-for-WordPress/web-stories-wp. The UI will look like this:

Screenshot 2020-06-16 at 18 36 26

If you have an account there, I can add you.

The [GitHub integration](https://docs.percy.io/v1/docs/github) is still WIP.

**Comment by swissspidy** [comment]
It seems to work!

See Percy build: https://percy.io/Web-Stories-for-WordPress/web-stories-wp/builds/5685961

The problem now is that it warns about missing snapshots from the integration tests.

**Comment by dvoytenko** [comment]
@swissspidy 

> The problem now is that it warns about missing snapshots from the integration tests.

Hmm. Isn't it a big problem though? Maybe we can work it around via a second project on Percy?

**Comment by swissspidy** [comment]
The message on Percy doesn’t make it sound like it's a big deal, it just leaves the missing screenshots untouched it seems.

But yeah a second project is probably better. Especially since we‘re gonna split up the repos anyway eventually.

**Comment by swissspidy** [comment]
We'd have to confirm whether two projects can be connected to the same repository though when we want to use https://github.com/marketplace/percy

**Review by spacedmonkey — APPROVED** [review]
I can't validate that screenshots are taken. But change seems valid.

**Comment by swissspidy** [comment]
New project: https://percy.io/google/web-stories-wp-e2e

**Comment by swissspidy** [comment]
Seems like this causes issues with installing Chromium due to different Puppeteer versions...

File: tests/e2e/specs/editor/media/insertMediaFromLibrary.js [hunk 1]
```diff
@@ -33,5 +38,7 @@ describe('Inserting Media from Media Library', () => {
     await expect(page).toMatchElement(
       '[data-testid="frameElement"]:nth-of-type(2)'
     );
+
+    await percySnapshot(page, 'E2E: Inserting Media from Media Library');
```
Comment (dvoytenko) [inline tests/e2e/specs/editor/media/insertMediaFromLibrary.js]: I don't think you need to use "E2E: " prefix everywhere, not that you're on a different project.

**Review by dvoytenko — APPROVED** [review]

File: tests/e2e/specs/editor/media/insertMediaFromLibrary.js (hunk 1 above)
Comment (swissspidy) [reply tests/e2e/specs/editor/media/insertMediaFromLibrary.js]: Indeed 👍

**Pull request merged**